modules/apps/document-library/document-library-preview-pdf/src/main/java/com/liferay/document/library/preview/pdf/internal/portlet/action/EditPDFPreviewMVCActionCommand.java

```diff
@@ -0,0 +1,95 @@
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.document.library.preview.pdf.internal.portlet.action;
+
+import com.liferay.configuration.admin.constants.ConfigurationAdminPortletKeys;
+import com.liferay.document.library.preview.pdf.exception.PDFPreviewException;
+import com.liferay.document.library.preview.pdf.internal.configuration.admin.service.PDFPreviewManagedServiceFactory;
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+import com.liferay.portal.configuration.persistence.listener.ConfigurationModelListenerException;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.servlet.SessionErrors;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alicia García
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.INSTANCE_SETTINGS,
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.SITE_SETTINGS,
+		"javax.portlet.name=" + ConfigurationAdminPortletKeys.SYSTEM_SETTINGS,
+		"mvc.command.name=/instance_settings/edit_pdf_preview"
+	},
+	service = MVCActionCommand.class
+)
+public class EditPDFPreviewMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		String scope = ParamUtil.getString(actionRequest, "scope");
+
+		if (Validator.isNull(scope)) {
+			throw new PortalException("Unsupported scope: " + scope);
+		}
+
+		long scopePK = ParamUtil.getLong(actionRequest, "scopePK");
+
+		if ((scopePK == 0) &&
+			!scope.equals(
+				ExtendedObjectClassDefinition.Scope.SYSTEM.getValue())) {
+
+			throw new PortalException(
+				"Invalid scope primary key 0 for " + scope + " scope");
+		}
+
+		try {
+			_updatePDFPreview(actionRequest, scope, scopePK);
+		}
+		catch (ConfigurationModelListenerException | PDFPreviewException
+					exception) {
+
+			SessionErrors.add(actionRequest, exception.getClass(), exception);
+
+			actionResponse.sendRedirect(
+				ParamUtil.getString(actionRequest, "redirect"));
+		}
+	}
+
+	private void _updatePDFPreview(
+			ActionRequest actionRequest, String scope, long scopePK)
+		throws Exception {
+
+		_pdfPreviewManagedServiceFactory.updatePDFPreview(
+			ParamUtil.getLong(actionRequest, "maxNumberOfPages"), scope,
+			scopePK);
+	}
```

Comment:
This can be inlined now, as all logic is internal to the managed service factory.

Reply:
inlining!